### ISSUE TITLE
PixelLens errors when showing a large image array

### ISSUE THREAD

**Issue by rickgladwin** [srwi/PyCharm-PixelLens#132]
Summary:
- PixelLens is able to display a set of images (with an index along the right side of the window) for a 491.5 MB, 1200 images, .npy image set (the test set)
- PixelLens throws an error when trying to display a larger set of images (1.97 GB, 4800 images, 640x640) with the same dimensions and properties (the training set)
- PixelLens is able to display individual images from the training set, but not the whole set.

----

An exception occurred in the plugin.

### Environment:
- Plugin Version: 0.6.0
- PyCharm Version: 2026.1.4 (PyCharm)
- Operating System: Mac OS X 15.7.5

### Exception details:
```js
kotlinx.serialization.json.internal.JsonDecodingException: Expected start of the object '{', but had 'EOF' instead at path: $
JSON input: 
kotlinx.serialization.json.internal.JsonExceptionsKt.JsonDecodingException(JsonExceptions.kt:24)
kotlinx.serialization.json.internal.JsonExceptionsKt.JsonDecodingException(JsonExceptions.kt:32)
kotlinx.serialization.json.internal.AbstractJsonLexer.fail(AbstractJsonLexer.kt:588)
kotlinx.serialization.json.internal.AbstractJsonLexer.fail$default(AbstractJsonLexer.kt:586)
kotlinx.serialization.json.internal.AbstractJsonLexer.unexpectedToken(AbstractJsonLexer.kt:790)
kotlinx.serialization.json.internal.StringJsonLexer.consumeNextToken(StringJsonLexer.kt:77)
kotlinx.serialization.json.internal.StreamingJsonDecoder.beginStructure(StreamingJsonDecoder.kt:102)
com.github.srwi.pixellens.data.BatchDataPayload$$serializer.deserialize(BatchDataPayload.kt:26)
com.github.srwi.pixellens.data.BatchDataPayload$$serializer.deserialize(BatchDataPayload.kt:26)
kotlinx.serialization.json.internal.StreamingJsonDecoder.decodeSerializableValue(StreamingJsonDecoder.kt:69)
kotlinx.serialization.json.Json.decodeFromString(Json.kt:149)
com.github.srwi.pixellens.imageProviders.ImageProvider.getBatchData(ImageProvider.kt:309)
com.github.srwi.pixellens.imageProviders.ImageProvider.getBatchByExpression(ImageProvider.kt:31)
com.github.srwi.pixellens.debugger.DebugValueLoader.loadBatch(DebugValueWatcher.kt:20)
com.github.srwi.pixellens.actions.ViewAsImageAction$actionPerformed$1.run(ViewAsImageAction.kt:40)
com.intellij.openapi.progress.impl.CoreProgressManager.startTask(CoreProgressManager.java:561)
com.intellij.openapi.progress.impl.ProgressManagerImpl.startTask(ProgressManagerImpl.java:197)
com.intellij.openapi.progress.impl.CoreProgressManager.lambda$runProcessWithProgressAsynchronously$7(CoreProgressManager.java:612)
com.intellij.openapi.progress.impl.ProgressRunner.lambda$submit$4(ProgressRunner.java:284)
com.intellij.openapi.progress.ProgressManager.lambda$runProcess$0(ProgressManager.java:98)
com.intellij.openapi.progress.impl.CoreProgressManager.lambda$runProcess$1(CoreProgressManager.java:287)
com.intellij.platform.diagnostic.telemetry.helpers.TraceKt.use(trace.kt:44)
com.intellij.openapi.progress.impl.CoreProgressManager.lambda$runProcess$2(CoreProgressManager.java:286)
com.intellij.openapi.progress.impl.CoreProgressManager.lambda$executeProcessUnderProgress$14(CoreProgressManager.java:744)
com.intellij.openapi.progress.impl.CoreProgressManager.registerIndicatorAndRun(CoreProgressManager.java:819)
com.intellij.openapi.progress.impl.CoreProgressManager.computeUnderProgress(CoreProgressManager.java:775)
com.intellij.openapi.progress.impl.CoreProgressManager.executeProcessUnderProgress(CoreProgressManager.java:743)
com.intellij.openapi.progress.impl.ProgressManagerImpl.executeProcessUnderProgress(ProgressManagerImpl.java:87)
com.intellij.openapi.progress.impl.CoreProgressManager.runProcess(CoreProgressManager.java:267)
com.intellij.openapi.progress.ProgressManager.runProcess(ProgressManager.java:98)
com.intellij.openapi.progress.impl.ProgressRunner.lambda$submit$5(ProgressRunner.java:284)
com.intellij.openapi.progress.impl.ProgressRunner$ProgressRunnable.run(ProgressRunner.java:556)
com.intellij.openapi.progress.impl.ProgressRunner.lambda$launchTask$17(ProgressRunner.java:511)
com.intellij.concurrency.ThreadContext.installThreadContext(threadContext.kt:305)
com.intellij.openapi.progress.impl.ProgressRunner.lambda$launchTask$18(ProgressRunner.java:510)
com.intellij.util.concurrency.ChildContext$runInChildContext$1.invoke(propagation.kt:192)
com.intellij.util.concurrency.ChildContext$runInChildContext$1.invoke(propagation.kt:192)
com.intellij.util.concurrency.ChildContext.runInChildContext(propagation.kt:198)
com.intellij.util.concurrency.ChildContext.runInChildContext(propagation.kt:192)
com.intellij.openapi.progress.impl.ProgressRunner.lambda$launchTask$19(ProgressRunner.java:507)
java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1090)
java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:614)
java.base/java.util.concurrent.Executors$PrivilegedThreadFactory$1.run(Executors.java:649)
java.base/java.lang.Thread.run(Thread.java:1474)
```

### Steps to reproduce:
1. Load a .npy image set into the PyCharm debugger in a Jupyter notebook
2. Load the data into a variable `X_train` using `np.load()`
3. Launch the debugger in the Jupyter notebook
4. click on `X_train` in the debugger and select "View as Image (PixelLens)"

### Expected:
- image set loads in the PixelLens window

### Actual:
- "Loading image..." progress message displays for a few seconds, then an error window appears: "Unexpected Error. PixelLens encountered an unexpected error. If possible, please report this issue on GitHub."

### Workarounds:
- open the individual images in the array (works, but doesn't allow quick browsing of the dataset)

### Code snippet:
```python
X_train = np.load(f'data/X_train.npy')
```

**Comment by srwi** [comment]
The bug you saw should now be fixed in the 0.7.0 release. However for very large arrays like your 1.97Gb example it will still not work because of the limited PyCharm heap. I will need to implement lazy data fetching for that, but I'll track that in a separate ticket: #134
